### PR TITLE
13/01/2021, search product by name

### PR DESCRIPTION
**Pull request by haupt-2855**
## Related Tickets
- [#44779][(https://edu-redmine.sun-asterisk.vn/issues/44779)]

## WHAT (optional)
- Search product by name
Screenshot from 2022-01-13 11-51-30
Screenshot from 2022-01-13 11-52-51

### CODE REVIEW

**Comment by haupt-2855** [comment]
ready

**Review by hieupp-1207 — APPROVED** [review]

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -63,5 +63,35 @@
...
+            <span><%= t ".all_categories" %></span>
+            <i class="bi bi-arrow-down"></i>
+          </div>
+        </div>
+      </div>
+      <div class="col-lg-9">
+        <div class="hero__search">
+          <div class="hero__search__form">
+            <%= form_tag(products_path, method: :get) do %>
+              <%= text_field_tag :term,params[:search], placeholder: "What do you need?" %>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb]: I18n text em ơi

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -63,5 +63,35 @@
...
+              <button type="submit" class="site-btn">SEARCH</button>
+            <% end %>
+          </div>
+          <div class="hero__search__phone">
+            <div class="hero__search__phone__icon">
+              <i class="fa fa-phone"></i>
+            </div>
+            <div class="hero__search__phone__text">
+              <h5>+65 11.188.888</h5>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb]: răng chỗ ni có text hay rứa hè
cho vô settings em nhé

File: app/views/layouts/_header.html.erb (hunk 1 above)
Comment (haupt-2855) [reply app/views/layouts/_header.html.erb]: dạ chị

File: app/views/layouts/_header.html.erb (hunk 2 above)
Comment (haupt-2855) [reply app/views/layouts/_header.html.erb]: dạ chị

**Comment by haupt-2855** [comment]
ready

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -1,5 +1,11 @@
 class ProductsController < ApplicationController
   def index
-    @pagy, @products = pagy Product.all.newest, items: Settings.number.digit_12
+    products = if params[:term].present?
+                 Product.search(params[:term])
+               else
+                 Product
+               end.newest
```
Comment (vanvtt-0952) [inline app/controllers/products_controller.rb]:
```suggestion
    products = Product.search(params[:term]).newest
```
check param search nên để ở controller nhé

File: app/models/product.rb [hunk 4]
```diff
@@ -5,4 +5,5 @@ class Product < ApplicationRecord
 
   scope :newest, ->{order created_at: :desc}
   scope :limit_8, ->{limit Settings.number.digit_8}
+  scope :search, ->(search){where("name LIKE ?", "%#{search}%")}
```
Comment (vanvtt-0952) [inline app/models/product.rb]:
```suggestion
  scope :search, ->(search){where("name LIKE ?", "%#{search}%") if search.present?}
```

**Comment by haupt-2855** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented

**Comment by haupt-2855** [comment]
ready

**Pull request merged**